build_runner/lib/src/entrypoint/daemon.dart

```diff
@@ -56,9 +58,13 @@ class DaemonCommand extends BuildRunnerCommand {
           builderApplications,
           options,
         );
+        var server =
+            await AssetServer.run(builder.reader, packageGraph.root.name);
+        File(assetServerPortFilePath(workingDirectory))
+            .writeAsStringSync('${server.port}');
         await daemon.start(requestedOptions, builder, builder.changes);
         stdout.writeln(readyToConnectLog);
-        await daemon.onDone;
+        await daemon.onDone.then((_) => server.stop());
```

Comment:
can avoid the empty closure since we don't care about the result.

`await daemon.onDone.whenComplete(server.stop);`

Reply:
Didn't know about the `whenComplete` method. Neat!